### PR TITLE
Move "created" and "updated" from Core Properties to DID document metadata

### PR DESCRIPTION
**Pull request by peacekeeper**
Addresses #203 and #174.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/pull/365.html" title="Last updated on Aug 10, 2020, 1:59 PM UTC (6663b38)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/365/3cc2641...6663b38.html" title="Last updated on Aug 10, 2020, 1:59 PM UTC (6663b38)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -3518,6 +3492,30 @@ <h2>
 ]»
         </pre>
 
+        <p>
+The next example demonstrates a JSON-encoded metadata structure that might be
+used as <a href="#did-document-metadata-properties">DID document metadata</a> to
+describe timestamps of the DID document.
```
Comment (msporny) [inline index.html:3498]:
```suggestion
describe timestamps associated with the DID document.
```

File: index.html [hunk 2]
```diff
@@ -3518,6 +3492,30 @@ <h2>
...
+        </p>
+
+        <pre class="example" title="JSON-encoded DID document metadata example">
+{
+  "created": "2002-10-10T17:00:00Z",
```
Comment (msporny) [inline index.html:3503]: I'm wondering if we should use a date that is within the range of when the DID spec existed. So, perhaps using created dates after 2018 and update dates after 2020 would be more realistic? :)

**Comment by msporny** [comment]
Merge conflicts need to be resolved.

File: index.html (hunk 2 above)
Comment (peacekeeper) [reply index.html]: Updated the example dates in https://github.com/w3c/did-core/pull/365/commits/260972366213973017c1b8bdba2f49d250de7ed5.

**Comment by peacekeeper** [comment]
@msporny thanks, merge conflicts should be fixed now.

**Comment by peacekeeper** [comment]
> However, this PR does not address issue #174 - as it does not specify the semantics of these values.

Thanks @selfissued, the PR says the following:

> DID document metadata SHOULD include an updated property **to indicate the timestamp of the last Update operation.**

I think **"timestamp of the last Update operation"** should be sufficient as a definition. My idea is that if additional explanations are needed, then those should be added to the section about the [Update operation](https://w3c.github.io/did-core/#update), not to the metadata property.

**Comment by msporny** [comment]
Mostly editorial, multiple positive reviews, changes requested and made, merging.

**Pull request merged**